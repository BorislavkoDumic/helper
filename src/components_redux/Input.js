import React, { Component } from "react";
import Radios from "./Radios";
import {
  addField,
  removeField,
  changeType,
  changeValue,
  changeLabel,
  saveValues,
  cancel
} from "../actions/index";
import { connect } from "react-redux";

class Input extends Component {
  changeType = value => {
    this.props.changeType(value);
  };
  createUI = options => {
    return options.map((inputField, i) => (
      <div key={i}>
        <label>Value</label>
        <input
          style={{ width: "240px" }}
          className="input-option"
          type="text"
          onChange={this.changeValue(i)}
          value={inputField}
        />
        <input
          type="button"
          className="button"
          value="Delete"
          onClick={this.removeField(i)}
        />
      </div>
    ));
  };
  changeValue = i => e => {
    this.props.changeValue(e.target.value, i);
  };
  changeLabel = e => {
    this.props.changeLabel(e.target.value);
  };
  addField = e => {
    e.preventDefault();
    this.props.addField();
  };
  removeField = i => e => {
    e.preventDefault();
    this.props.removeField(i);
  };
  saveValues = valuesString => {
    this.props.saveValues(valuesString);
    this.props.changeTextfield(valuesString);
  };
  cancel = () => {
    this.props.closeForm();
  };
  render() {
    const { options, type, label } = this.props;
    const uiElements = this.createUI(options);

    let jsonObject = {
      label: label,
      type: type
    };

    if (jsonObject.type === "select" || jsonObject.type === "radios") {
      jsonObject = {
        ...jsonObject,
        options: options
      };
    }

    const valuesString = JSON.stringify(jsonObject);

    return (
      <div>
        <div>
          <div className="ui-dialog-titlebar ui-corner-all ui-widget-header ui-helper-clearfix">
            <span
              className="dialog-title"
              style={{ fontSize: 15, color: "white" }}
            >
              Fill the form
            </span>
            <button
              title="CLose"
              className="ui-button ui-corner-all ui-widget ui-button-icon-only ui-dialog-titlebar-close"
              onClick={this.cancel}
            >
              <span className="ui-button-icon ui-icon ui-icon-closethick"></span>
              <span className="ui-button-icon-space"></span>
            </button>
          </div>
          <div
            className="ui-front ui-dialog-content ui-widget-content"
            style={{ width: "auto" }}
          >
            {this.props.errors.map((item, i) => {
              return (
                <div key={i}>
                  <span style={{ color: "red", fontSize: 15 }}>{item}</span>
                </div>
              );
            })}
            <label className="label">
              Label:
              <input
                type="text"
                onChange={this.changeLabel}
                value={label}
                style={{ width: "-webkit-fill-available" }}
              ></input>
            </label>
            <div className="inputForm">
              <label>Type</label>
              <Radios
                name="type"
                value={type}
                options={[
                  {
                    value: "text",
                    label: "Textfiled"
                  },
                  {
                    value: "checkbox",
                    label: "Checkbox"
                  },
                  {
                    value: "select",
                    label: "Select"
                  },
                  {
                    value: "radios",
                    label: "Radios"
                  }
                ]}
                onChange={this.changeType}
              />
              {type === "select" || type === "radios" ? (
                <div>
                  <form
                    className="options-form"
                    style={{ marginBottom: "10px" }}
                  >
                    <button
                      style={{ marginLeft: "30%" }}
                      className="button button--primary js-form-submit form-submit ui-button ui-corner-all ui-widget"
                      onClick={this.addField}
                    >
                      Add more options
                    </button>
                    {uiElements}
                  </form>
                </div>
              ) : null}
            </div>
            <textarea
              onChange={changeValue}
              className="text-area"
              value={valuesString}
              style={{ width: "-webkit-fill-available", height: "50px" }}
            ></textarea>
          </div>
          <div className="ui-dialog-buttonpane ui-widget-content ui-helper-clearfix">
            <button
              type="button"
              className="button button--primary js-form-submit form-submit ui-button ui-corner-all ui-widget"
              onClick={() => this.saveValues(valuesString)}
            >
              Save
            </button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    type: state.inputFields.type,
    options: state.inputFields.options,
    label: state.inputFields.label,
    errors: state.inputFields.errors
  };
};

const mapDispatchToProps = dispatch => {
  return {
    changeType: value => dispatch(changeType(value)),
    addField: () => dispatch(addField()),
    removeField: i => dispatch(removeField(i)),
    changeValue: (value, index) => dispatch(changeValue(value, index)),
    changeLabel: label => dispatch(changeLabel(label)),
    saveValues: valuesString => dispatch(saveValues(valuesString)),
    cancel: () => dispatch(cancel())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Input);
