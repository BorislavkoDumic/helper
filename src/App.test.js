import React from "react";
import Input from "./components_redux/Input";
import { createStore } from "redux";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import reducer from "./reducers";
import {
  openOverlay,
  changeLabel,
  changeType,
  addField,
  changeValue,
  removeField,
  saveValues,
  cancel
} from "./actions/index";

describe("Open overlay and input value of label", () => {
  it("Matches the snapshot", () => {
    const store = createStore(reducer);
    store.dispatch(openOverlay());
    store.dispatch(changeLabel("Label value"));

    const tree = renderer
      .create(
        <Provider store={store}>
          <Input />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("Selecting the type of radio buttons", () => {
  it("Matches the snapshot ", () => {
    const store = createStore(reducer);
    store.dispatch(openOverlay());
    store.dispatch(changeLabel("Label value text"));
    store.dispatch(changeType("text"));

    const tree = renderer
      .create(
        <Provider store={store}>
          <Input />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Matches the snapshot ", () => {
    const store = createStore(reducer);
    store.dispatch(openOverlay());
    store.dispatch(changeLabel("Label value checkbox"));
    store.dispatch(changeType("checkbox"));

    const tree = renderer
      .create(
        <Provider store={store}>
          <Input />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Matches the snapshot ", () => {
    const store = createStore(reducer);
    store.dispatch(openOverlay());
    store.dispatch(changeLabel("Label value select"));
    store.dispatch(changeType("select"));

    const tree = renderer
      .create(
        <Provider store={store}>
          <Input />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Matches the snapshot ", () => {
    const store = createStore(reducer);
    store.dispatch(openOverlay());
    store.dispatch(changeLabel("Label value radios"));
    store.dispatch(changeType("radios"));

    const tree = renderer
      .create(
        <Provider store={store}>
          <Input />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("Selecting the type of radio buttons (select), add fields and input values", () => {
  it("Matches the snapshot ", () => {
    const store = createStore(reducer);
    store.dispatch(openOverlay());
    store.dispatch(changeLabel("Label value"));
    store.dispatch(changeType("select"));
    store.dispatch(addField());
    store.dispatch(addField());
    store.dispatch(addField());
    store.dispatch(changeValue("1", 0));
    store.dispatch(changeValue("2", 1));
    store.dispatch(changeValue("3", 2));

    const tree = renderer
      .create(
        <Provider store={store}>
          <Input />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe("Selecting the type of radio buttons (radios), add fields and input values then change the type (text)", () => {
    it("Matches the snapshot ", () => {
      const store = createStore(reducer);
      store.dispatch(openOverlay());
      store.dispatch(changeLabel("Label value"));
      store.dispatch(changeType("radios"));
      store.dispatch(addField());
      store.dispatch(addField());
      store.dispatch(addField());
      store.dispatch(changeValue("1", 0));
      store.dispatch(changeValue("2", 1));
      store.dispatch(changeValue("3", 2));
      store.dispatch(changeType("text"));

      const tree = renderer
        .create(
          <Provider store={store}>
            <Input />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("Selecting the type of radio buttons (select), add fields and input values then remove one field", () => {
    it("Matches the snapshot ", () => {
      const store = createStore(reducer);
      store.dispatch(openOverlay());
      store.dispatch(changeLabel("Label value"));
      store.dispatch(changeType("select"));
      store.dispatch(addField());
      store.dispatch(addField());
      store.dispatch(addField());
      store.dispatch(changeValue("1", 0));
      store.dispatch(changeValue("2", 1));
      store.dispatch(changeValue("3", 2));
      store.dispatch(removeField(1));

      const tree = renderer
        .create(
          <Provider store={store}>
            <Input />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("Selecting the type of radio buttons (select), add fields and input values then click save button", () => {
    it("Matches the snapshot ", () => {
      const store = createStore(reducer);
      store.dispatch(openOverlay());
      store.dispatch(changeLabel("Label value"));
      store.dispatch(changeType("select"));
      store.dispatch(addField());
      store.dispatch(addField());
      store.dispatch(addField());
      store.dispatch(changeValue("1", 0));
      store.dispatch(changeValue("2", 1));
      store.dispatch(changeValue("3", 2));
      store.dispatch(
        saveValues({
          label: "Label Value",
          type: "select",
          options: ["1", "2", "3"]
        })
      );

      const tree = renderer
        .create(
          <Provider store={store}>
            <Input />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("Selecting the type of radio buttons (select), add fields and input values then click cancel button", () => {
    it("Matches the snapshot ", () => {
      const store = createStore(reducer);
      store.dispatch(openOverlay());
      store.dispatch(changeLabel("Label value"));
      store.dispatch(changeType("select"));
      store.dispatch(addField());
      store.dispatch(addField());
      store.dispatch(addField());
      store.dispatch(changeValue("1", 0));
      store.dispatch(changeValue("2", 1));
      store.dispatch(changeValue("3", 2));
      store.dispatch(cancel());

      const tree = renderer
        .create(
          <Provider store={store}>
            <Input />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
