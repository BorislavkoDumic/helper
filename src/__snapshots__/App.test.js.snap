// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced tests of app  Selecting the type of radio buttons (radios), add fields and input values then change the type (text) 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value="Label value"
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\"Label value\\",\\"type\\":\\"text\\"}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`Advanced tests of app  Selecting the type of radio buttons (select), add fields and input values 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value="Label value"
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
      <div>
        <label>
          ADD MORE OPTIONS
        </label>
        <form
          className="options-form"
        >
          <button
            className="button-option"
            onClick={[Function]}
          >
            ADD
          </button>
          <div>
            <label>
              Value
            </label>
            <input
              className="input-option"
              onChange={[Function]}
              type="text"
              value="1"
            />
            <input
              className="button-option"
              onClick={[Function]}
              type="button"
              value="DELETE"
            />
          </div>
          <div>
            <label>
              Value
            </label>
            <input
              className="input-option"
              onChange={[Function]}
              type="text"
              value="2"
            />
            <input
              className="button-option"
              onClick={[Function]}
              type="button"
              value="DELETE"
            />
          </div>
          <div>
            <label>
              Value
            </label>
            <input
              className="input-option"
              onChange={[Function]}
              type="text"
              value="3"
            />
            <input
              className="button-option"
              onClick={[Function]}
              type="button"
              value="DELETE"
            />
          </div>
        </form>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\"Label value\\",\\"type\\":\\"select\\",\\"options\\":[\\"1\\",\\"2\\",\\"3\\"]}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`Advanced tests of app  Selecting the type of radio buttons (select), add fields and input values then click cancel button 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value="Click here to for input text"
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value=" "
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
      <div>
        <label>
          ADD MORE OPTIONS
        </label>
        <form
          className="options-form"
        >
          <button
            className="button-option"
            onClick={[Function]}
          >
            ADD
          </button>
        </form>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\" \\",\\"type\\":\\"select\\",\\"options\\":[]}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`Advanced tests of app  Selecting the type of radio buttons (select), add fields and input values then click save button 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value={
        Object {
          "label": "Label Value",
          "options": Array [
            "1",
            "2",
            "3",
          ],
          "type": "select",
        }
      }
    />
  </div>
</div>
`;

exports[`Advanced tests of app  Selecting the type of radio buttons (select), add fields and input values then remove one field 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value="Label value"
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
      <div>
        <label>
          ADD MORE OPTIONS
        </label>
        <form
          className="options-form"
        >
          <button
            className="button-option"
            onClick={[Function]}
          >
            ADD
          </button>
          <div>
            <label>
              Value
            </label>
            <input
              className="input-option"
              onChange={[Function]}
              type="text"
              value="1"
            />
            <input
              className="button-option"
              onClick={[Function]}
              type="button"
              value="DELETE"
            />
          </div>
          <div>
            <label>
              Value
            </label>
            <input
              className="input-option"
              onChange={[Function]}
              type="text"
              value="3"
            />
            <input
              className="button-option"
              onClick={[Function]}
              type="button"
              value="DELETE"
            />
          </div>
        </form>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\"Label value\\",\\"type\\":\\"select\\",\\"options\\":[\\"1\\",\\"3\\"]}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`Basic test fof showing overlay Open overlay and input value of label 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value="Label value"
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\"Label value\\",\\"type\\":\\"\\"}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`Selecting the type of radio buttons Select type checkbox 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value="Label value checkbox"
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\"Label value checkbox\\",\\"type\\":\\"checkbox\\"}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`Selecting the type of radio buttons Select type radios 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value="Label value radios"
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
      <div>
        <label>
          ADD MORE OPTIONS
        </label>
        <form
          className="options-form"
        >
          <button
            className="button-option"
            onClick={[Function]}
          >
            ADD
          </button>
        </form>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\"Label value radios\\",\\"type\\":\\"radios\\",\\"options\\":[]}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`Selecting the type of radio buttons Select type select 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value="Label value select"
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
      <div>
        <label>
          ADD MORE OPTIONS
        </label>
        <form
          className="options-form"
        >
          <button
            className="button-option"
            onClick={[Function]}
          >
            ADD
          </button>
        </form>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\"Label value select\\",\\"type\\":\\"select\\",\\"options\\":[]}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`Selecting the type of radio buttons Select type text 1`] = `
<div>
  <div
    className="input-conntainer"
  >
    <input
      className="input-field"
      onClick={[Function]}
      placeholder="Click here to for input text"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="overlay"
    style={
      Object {
        "width": 500,
      }
    }
  >
    <h2>
      Fill the form
    </h2>
    <label
      className="label"
    >
      Label:
      <input
        onChange={[Function]}
        type="text"
        value="Label value text"
      />
    </label>
    <div
      className="inputForm"
    >
      <div>
        Type
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="text"
            name="type"
            onChange={[Function]}
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Textfiled
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="checkbox"
            name="type"
            onChange={[Function]}
            type="radio"
            value="checkbox"
          />
          <label
            htmlFor="checkbox"
          >
            CheckBox
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="select"
            name="type"
            onChange={[Function]}
            type="radio"
            value="select"
          />
          <label
            htmlFor="select"
          >
            Select
          </label>
        </div>
        <div
          className="inputForm"
        >
          <input
            checked={false}
            id="radios"
            name="type"
            onChange={[Function]}
            type="radio"
            value="radios"
          />
          <label
            htmlFor="radios"
          >
            Radios
          </label>
        </div>
      </div>
    </div>
    <textarea
      className="text-area"
      onChange={[Function]}
      value="{\\"label\\":\\"Label value text\\",\\"type\\":\\"text\\"}"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;
